lib/rules/font-family-name-quotes/index.js

```diff
@@ -56,8 +56,67 @@ function quotesRequired(family) {
 		.some((word) => /^(?:-?\d|--)/.test(word) || !/^[-\w\u{00A0}-\u{10FFFF}]+$/u.test(word));
 }
 
+/**
+ * @typedef {{
+ * name: string,
+ * hasQuotes: boolean,
+ * sourceIndex: number,
+ * removeQuotes: function,
+ * addQuotes: function
+ * }} MutableNode
+ */
+
+/**
+ *
+ * @param {import('postcss-value-parser').Node[]} fontFamilies
+ * @param {import('postcss').Declaration} decl
+ */
+const MakeMutableFontFamilies = (fontFamilies, decl) => {
+	/**
+	 * @type {MutableNode[]}
+	 */
+	const mutableNodes = [];
+
+	fontFamilies.forEach((fontFamily, idx) => {
+		const newNode = {
+			name: fontFamily.value,
+			sourceIndex: fontFamily.sourceIndex,
+			hasQuotes: 'quote' in fontFamily,
+			resetIndexes: (/** @type {number} */ offset) => {
+				mutableNodes.slice(idx + 1).forEach((n) => (n.sourceIndex += offset));
+			},
+			removeQuotes() {
+				if (this.hasQuotes === false) return;
+
+				const openIndex = this.sourceIndex;
+				const closeIndex = openIndex + this.name.length + 2;
+
+				this.hasQuotes = true;
```

Comment:
[ask] Is `false` valid?

```suggestion
				this.hasQuotes = false;
```